docs/guides/streaming/sources/google-cloud-pubsub.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: "Google Cloud PubSub"
+description:
+  "Ingest data from Google Cloud PubSub event streaming sources."
+---
+
+<img
+  alt="Google Cloud PubSub Source"
+  src="https://lh3.googleusercontent.com/1-WPjO3QIWh0PKPXhHNcKUNcTCeKX218WsC6nIv1xGjGg3VQci-nG9uTqUyBlStJ8YR5LVaC8QFXN6Y7Y9lz=w160-h160"
+/>
+
+<br />
+
+## Config
+
+```yaml
+connector_type: google_cloud_pubsub
+project_id: test-project-id
+topic_id: test-topic-id
+subscription_id: test-subscription-id-0
+timeout: 5
+batch_size: 100
+pubsub_emulator_host: "host.docker.internal:8085"
+google_application_credentials: str = None  # e.g., "./google_credentials.json"
+```
+
+## Set up Google Cloud Pubsub emulator
```

Comment:
Let's move the content of emulator to the bottom of the doc since most of the users don't need to use emulator.

Reply:
Done